drivers/rtc/Kconfig.mc146818

```diff
@@ -0,0 +1,9 @@
+# Intel SoC RTC configuration options
+
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config RTC_MOTOROLA_MC146818
+	bool "RTC driver for x86 CMOS/RTC clock"
+	default y
+	depends on DT_HAS_MOTOROLA_MC146818_ENABLED && SOC_RAPTOR_LAKE
```

Comment:
Why did you add the SoC dependency here? I don’t think that’s the right thing to do. What are you trying to solve with that?

Reply:
> Why did you add the SoC dependency here? I don’t think that’s the right thing to do. What are you trying to solve with that?

If this is not added, while running rtc tests for other boards with just counter instance and compatible "motorola,mc146818"  status okay, is building rtc driver without instance and failing.

Reply:
Other way is that if we have counter instance with compatible "motorola,mc164818" in dts then we can add RTC instance for those boards also.

Reply:
Well, whatever the solution, it can't be that we restrict using the driver to only one specific SoC.

> If this is not added, while running rtc tests for other boards with just counter instance and compatible "motorola,mc146818" status okay, is building rtc driver without instance and failing.

Would one solution for this be to introduce a `supported: rtc` property to the board yaml files, and then we enable RTC tests only if the board indicates support for RTC? Any comments from @bjarki-trackunit ?

Reply:
depends_on: rtc is not there for test/drivers/rtc/rtc_api_helpers/tetscase.yaml file. It is failing for all x86 boards which doesn't have an rtc instance in dts but motorola,mc146818 compatible is okay status. Build of rpl_api_helpers test application is failing on qemu_x86 on CI.

Reply:
I didn't add supported: rtc in rpl_crb board yaml file. I will add it.